sdks/java/core/src/main/java/org/apache/beam/sdk/values/PDone.java

```diff
@@ -37,12 +44,25 @@ public static PDone in(Pipeline pipeline) {
   }
 
   @Override
+  public Pipeline getPipeline() {
+    return pipeline;
+  }
+
+  /**
+   * A {@link PDone} contains no {@link PValue PValues}.
+   */
+  @Override
   public Map<TupleTag<?>, PValue> expand() {
-    // A PDone contains no PValues.
     return Collections.emptyMap();
   }
 
-  private PDone(Pipeline pipeline) {
-    super(pipeline);
-  }
+  /** Does nothing; there are no concrete outputs to record. */
+  @Override
+  public void recordAsOutput(AppliedPTransform<?, ?, ?> transform) {}
```

Comment:
It seems to me that it is the applied PTransform that is needed, so the resulting API will need to be somewhere in between the two. So, since I can't give actionable advice in a `@deprecated` comment, let's just let the bug track our intention to change this.